### DISCUSSION TITLE
Further documentation cleanup

### DISCUSSION THREAD

**Discussion by onpikono** [dotnet/fsharp#12316] [Ideas]
I am not an experienced developer; however, I would like to contribute however I can. I was thinking about going through the refreshened example documentation and doing the following:

- Minor adjustment to text, such as adding constraints to function descriptions ("...takes a *non-empty* list...") to make it a bit clearer what they are even before talking about exceptions, unifying of used terms etc.
- Making examples consistent. Currently some results of evaluation are in form of a code comment, some are in a separate code block, some are as inline code. The same goes for how exceptions are described.
- Ensuring all functions have their "corner cases" (exceptions) documented
- Ensuring all code samples, including evaluation results, are properly formatted (fantomas 4.6.0), i.e. in line with [F# code style guide](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting)
- We might even define a standard for example documentation based on what we come to as the optimal structure and formatting.

I would appreciate any feedback to the above "plan" before I start working on it.

Thank you!

**Comment by dsyme** [comment]
@onpikono These contributions would be very much appreciated!

**Reply by onpikono** [reply]
Ok, I'll give it a go.
